src/qibolab/pulses.py

```diff
@@ -476,7 +464,7 @@ def envelope_waveform_i(self) -> Waveform:
             waveform.serial = f"Envelope_Waveform_I(num_samples = {num_samples}, amplitude = {format(self.pulse.amplitude, '.6f').rstrip('0').rstrip('.')}, shape = {repr(self)})"
             return waveform
         else:
```

Comment:
Not sure Pylint hasn't flagged, but: do you really need the `else`? The line above is `return`ing